samples/bot-batch-operations/csharp/Bots/TeamsBatchOperationsBot.cs

```diff
@@ -0,0 +1,427 @@
+
+            await turnContext.SendActivityAsync(MessageFactory.Text(message), cancellationToken);
+        }
+
+        private Attachment GetInputCard(ITurnContext<IMessageActivity> turnContext, string resourceName, string operationName)
+        {
+            string[] filepath = { ".", "Resources", resourceName };
+
+            var adaptiveCardJson = File.ReadAllText(Path.Combine(filepath));
+            if (operationName == "team-id")
+                adaptiveCardJson = adaptiveCardJson.Replace("the operation", "the team");
+            adaptiveCardJson = adaptiveCardJson.Replace("\"inputid", $"\"{operationName}");
+            
+            AdaptiveCardTemplate template = new AdaptiveCardTemplate(adaptiveCardJson);
+            var payloadData = new
+            {
+                createdBy = turnContext.Activity.From.Name
+            };
+
+            //"Expand" the template -this generates the final Adaptive Card payload
+            var cardJsonstring = template.Expand(payloadData);
+            var adaptiveCardAttachment = new Attachment()
+            {
+                ContentType = "application/vnd.microsoft.card.adaptive",
+                Content = JsonConvert.DeserializeObject(cardJsonstring),
+            };
+
+            return adaptiveCardAttachment;
+        }
+
+        private async Task SendDataToBatchOperations(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var operation = turnContext.Activity.Value.ToString();
+
+            if (operation.Contains("user"))
+                await MessageListOfUsersAsync(turnContext, cancellationToken);
+            else if (operation.Contains("channel"))
+                await MessageListOfChannelsAsync(turnContext, cancellationToken);
+            else if (operation.Contains("team"))
+                await MessageAllUsersInTeamAsync(turnContext, cancellationToken);
+            else if (operation.Contains("state"))
+                await GetOperationStateAsync(turnContext, cancellationToken);
+            else if (operation.Contains("entries"))
+                await GetFailedEntriesAsync(turnContext, cancellationToken);
+            else if (operation.Contains("cancel"))
+                await CancelOperationAsync(turnContext, cancellationToken);
+        }
+       
+        private string GetId(object source, string propName)
+        {
+            var jobject = source as JObject;
+            var jtoken = jobject?[propName];
+
+            return jtoken?.Value<string>().Trim();
+        }
+
+        private static async Task SendWelcomeCard(ITurnContext<IMessageActivity> turnContext, HeroCard card, CancellationToken cancellationToken)
+        {
+            card.Title = "Welcome!";
+
+            var activity = MessageFactory.Attachment(card.ToAttachment());
+
+            await turnContext.SendActivityAsync(activity, cancellationToken);
+        }
+
+        //-----Subscribe to Conversation Events in Bot integration
```

Comment:
We can remove all these methods (from here to the end of the file) because this sample is not meant to show these functionalities.